tests/test_paperqa.py

```diff
@@ -766,18 +771,39 @@ def test_hybrid_embedding(stub_data_dir: Path, vector_store: type[VectorStore])
 
 
 def test_custom_llm(stub_data_dir: Path) -> None:
-    from llmclient import Chunk
-
     class StubLLMModel(LLMModel):
         name: str = "myllm"
 
-        async def acomplete(self, prompt: str) -> Chunk:  # noqa: ARG002
-            return Chunk(text="Echo", prompt_tokens=1, completion_tokens=1)
+        async def acompletion(
+            self, messages: list[Message], **kwargs  # noqa: ARG002
+        ) -> list[LLMResult]:
+            return [
+                LLMResult(
+                    model=self.name,
+                    text="Echo",
+                    prompt=messages,
+                    prompt_count=1,
+                    completion_count=1,
+                )
+            ]
+
+        @rate_limited
+        async def acompletion_iter(
+            self, messages: list[Message], **kwargs  # noqa: ARG002
+        ) -> AsyncIterable[LLMResult]:
+            yield LLMResult(
+                model=self.name,
+                text="Echo",
+                prompt=messages,
+                prompt_count=1,
+                completion_count=1,
+            )
 
-        async def acomplete_iter(
-            self, prompt: str  # noqa: ARG002
-        ) -> AsyncIterable[Chunk]:
-            yield Chunk(text="Echo", prompt_tokens=1, completion_tokens=1)
+        async def check_rate_limit(
+            self, token_count: float, **kwargs  # noqa: ARG002
+        ) -> None:
+            # Implements a dummy rate limit checker.
+            return
```

Comment:
```suggestion
        ) -> None:
            """This is a dummy check."""
```

No need for `return`